app/slashcommands/command_remote.go

```diff
@@ -97,39 +104,98 @@ func (rp *RemoteProvider) GetAutoCompleteListItems(a *app.App, commandArgs *mode
 	return nil, fmt.Errorf("`%s` is not a dynamic argument", arg.Name)
 }
 
-func (rp *RemoteProvider) doAdd(a *app.App, args *model.CommandArgs, margs map[string]string) *model.CommandResponse {
+// doInvite creates and displays an encrypted invite that can be used by a remote site to establish a simple trust.
+func (rp *RemoteProvider) doInvite(a *app.App, args *model.CommandArgs, margs map[string]string) *model.CommandResponse {
+	password := margs["password"]
+	if password == "" {
+		return responsef("Missing or empty `password`")
+	}
+
 	name := margs["name"]
 	if name == "" {
 		return responsef("Missing or empty `name`")
 	}
 
-	host := margs["host"]
-	if host == "" {
-		return responsef("Missing or empty `host`")
+	rc := &model.RemoteCluster{
+		DisplayName: name,
+		Token:       model.NewId(),
+	}
+
+	rcSaved, appErr := a.AddRemoteCluster(rc)
+	if appErr != nil {
+		return responsef("Could not add remote cluster: %v", appErr)
+	}
+
+	url := a.GetSiteURL()
+	if url == "" {
+		return responsef("SiteURL not set. Please set this via the system console.")
 	}
 
-	port := margs["port"]
-	if port == "" {
-		port = "8065"
+	// Display the encrypted invitation
+	invite := &model.RemoteClusterInvite{
+		RemoteId: rcSaved.RemoteId,
+		SiteURL:  url,
+		Token:    rcSaved.Token,
 	}
-	iport, err := strconv.ParseInt(port, 10, 32)
+	encrypted, err := invite.Encrypt(password)
 	if err != nil {
-		return responsef("Invalid `port`: %v", err)
+		return responsef("Could not create invitation: %v", err)
```

Comment:
Should we be more specific? e.g. "Could not encrypt invitation".

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/16432/commits/c785b4b692a65411487822c2ce7c40f614bdd121